### DISCUSSION TITLE
Learning Rust in 2024

### DISCUSSION THREAD

**Discussion by pretzelhammer** [pretzelhammer/rust-blog#84] [General]
General discussion thread for the article [Learning Rust in 2024](https://github.com/pretzelhammer/rust-blog/blob/master/posts/learning-rust-in-2024.md).

Feel free to leave comments, questions, and suggestions here!

**Comment by GNUqb114514** [comment]
I'm going to translate this. :)
Just for an FYI: See also [course.rs](https://course.rs/about-book.html) for Chinese content.

**Comment by b9Joker108** [comment]
I just had a glance, approaching a deep graze, over this weblog post of yours. My current principal coding device is the native Termux host environment and any number of proot chroots in a Termux GNU/Linux userland on an unrooted Samsang Galaxy Tab S9 Ultra Android device and I have had `rustup`, `rust`, `cargo` and `rustc` all installed and working in my Termux host environment at different times, but no matter what I try, I cannot realise a working environment of `rustup`, `cargo` and `rustc` at the same time. I am interested if you have any insight on this: https://users.rust-lang.org/t/trouble-with-rust-ecosystem-and-toolchain-in-termux/122212 

BTW, I really hoped to see a section on setting up an optimal rust environment and toolchain for coding and projects, but could not see one outlined in your weblog. I feel this is very important and needed. 

Thank you so much for your endeavour. Your weblog is wonderful!
